lib/mpl_toolkits/axes_grid1/inset_locator.py

```diff
@@ -1,50 +1,77 @@
 from matplotlib.path import Path
+from matplotlib.transforms import Bbox, BboxTransformTo
+from matplotlib.transforms import IdentityTransform, TransformedBbox
 
-from matplotlib.patches import Rectangle
+from . import axes_size as Size
+from .parasite_axes import HostAxes
 
 
 class InsetPosition(object):
+    @docstring.dedent_interpd
     def __init__(self, parent, lbwh):
+        """
+        An object for positioning an inset axes.
+
+        This is created by specifying the normalized coordinates in the axes,
+        instead of the figure.
+
+        Parameters
+        ----------
+        parent : `matplotlib.axes.Axes`
+            Axes to use for normalizing coordinates.
+
+        lbwh : iterable of four floats
+            The left edge, bottom edge, width, and height of the inset axes, in
+            units of the normalized coordinate of the *parent* axes.
+
+        See Also
+        --------
+        :meth:`matplotlib.axes.Axes.set_axes_locator`
+
+        Examples
+        --------
+        The following bounds the inset axes to a box with 20%% of the parent
+        axes's height and 40%% of the width. The size of the axes specified
+        ([0, 0, 1, 1]) ensures that the axes completely fills the bounding box:
+
+        >>> parent_axes = plt.gca()
+        >>> ax_ins = plt.axes([0, 0, 1, 1])
+        >>> ip = InsetPosition(ax, [0.5, 0.1, 0.4, 0.2])
+        >>> ax_ins.set_axes_locator(ip)
+        """
         self.parent = parent
-        self.lbwh = lbwh  # position of the inset axes in
-                          # the normalized coordinate of the parent axes
+        self.lbwh = lbwh
 
     def __call__(self, ax, renderer):
         bbox_parent = self.parent.get_position(original=False)
-        trans = mtrans.BboxTransformTo(bbox_parent)
-        bbox_inset = mtrans.Bbox.from_bounds(*self.lbwh)
-        bb = mtrans.TransformedBbox(bbox_inset, trans)
+        trans = BboxTransformTo(bbox_parent)
+        bbox_inset = Bbox.from_bounds(*self.lbwh)
+        bb = TransformedBbox(bbox_inset, trans)
         return bb
 
 
 class AnchoredLocatorBase(AnchoredOffsetbox):
     def __init__(self, bbox_to_anchor, offsetbox, loc,
                  borderpad=0.5, bbox_transform=None):
-
-        super(AnchoredLocatorBase, self).__init__(loc,
-                                                  pad=0., child=None,
-                                                  borderpad=borderpad,
-                                                 bbox_to_anchor=bbox_to_anchor,
-                                                 bbox_transform=bbox_transform)
+        super(AnchoredLocatorBase, self).__init__(
+            loc, pad=0., child=None, borderpad=borderpad,
+            bbox_to_anchor=bbox_to_anchor, bbox_transform=bbox_transform
+        )
 
     def draw(self, renderer):
         raise RuntimeError("No draw method should be called")
 
     def __call__(self, ax, renderer):
+        self.axes = ax
```

Comment:
Both `AnchoredZoomLocator` and `AnchoredSizeLocator` save the axes in `__call__` for use in `get_extent`. This was just an attempt at removing code duplication, although admittingly it isn't documented clearly for subclasses.